src/Raven.Server/Documents/Queries/Parser/QueryParser.cs

```diff
@@ -875,63 +875,83 @@ private static QueryExpression FinalProcessingOfMatchExpression(List<MatchPath>
             };
             return op;
         }
-
-        private static QueryExpression BreakPatternChainToAndClauses(List<MatchPath> list, out QueryExpression op)
+      
+        private static QueryExpression BreakPatternChainToAndClauses(List<MatchPath> matchPaths, out QueryExpression op)
         {
-            // need to split the expression based on the direction of the arrows
+            op = null;
             var clauses = new List<PatternMatchElementExpression>();
-            bool reversed = false;
-            for (int i = list.Count - 1; i >= 1; i--)
+
+            //the last path element cannot be a "cross-road"
+            while (matchPaths.Count > 1)
             {
-                if (list[i].EdgeType != list[i - 1].EdgeType)
+                var hasFoundJunction = false;
+                for (var i = 0; i < matchPaths.Count - 2; i++)
```

Comment:
The last item in the path cannot be the "junction"
Also, since we are dealing with current and next items and compare between their directions to detect "junctions", there is no need to handle the last one

Reply:
Can you add an assert or failure to check that it can't be, just to be safe?

Reply:
Sure, I will add more precautions to that area of code.